### ISSUE TITLE
calculate_loss 的计算尺度问题

### ISSUE THREAD

**Issue by hlee13** [dgliu/KDD23_EFIN#10]
calculate_loss中143行，uc、ut 默认都是elu激活后的 logits尺度（值域是正负无穷），y_true是目标值（概率尺度值域[0-1]），然后用的 MSE损失，是这样的吗

**Comment by hlee13** [comment]
看了paper，我还以为概率部分都用的 bce

**Comment by dgliu** [comment]
> calculate_loss中143行，uc、ut 默认都是elu激活后的 logits尺度（值域是正负无穷），y_true是目标值（概率尺度值域[0-1]），然后用的 MSE损失，是这样的吗

感谢你对我们工作的关注。

论文中的损失函数符号只是指代任意可选形式，在实验中你可以对bce和mse都进行尝试。只是我们在binary label的uplift基准集实验中，发现使用BCE很容易让重参数模型架构训练出现崩塌，因此我们最后与EUEN一样采用了MSE。

**Comment by hlee13** [comment]
> > calculate_loss中143行，uc、ut 默认都是elu激活后的 logits尺度（值域是正负无穷），y_true是目标值（概率尺度值域[0-1]），然后用的 MSE损失，是这样的吗
> 
> 感谢你对我们工作的关注。
> 
> 论文中的损失函数符号只是指代任意可选形式，在实验中你可以对bce和mse都进行尝试。只是我们在binary label的uplift基准集实验中，发现使用BCE很容易让重参数模型架构训练出现崩塌，因此我们最后与EUEN一样采用了MSE。

谢谢你的答复

**Comment by BITQinYong** [comment]
loss2 里面，用的是bce，但是直接用t_logits，也没有过sigmoid，这是为什么

**Comment by dgliu** [comment]
> loss2 里面，用的是bce，但是直接用t_logits，也没有过sigmoid，这是为什么

torch.nn.BCEWithLogitsLoss接收logits作为输入，与BCELoss有所不同，请参照其官方文档说明：https://docs.pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html

**Comment by BITQinYong** [comment]
还有一个问题：
代码中：对于x_rep 进行self_attn之前，会对它进行L2 norm
如果随机torch.randn() 一个随意n*dim 的张量，过完self_attn之后，n个张量的值都是一样的，乃至最终输出的c_logits都是一样的。
这是因为你的输入值是大于1的float，我记得那个公开是数据集好像是，你才做L2 norm的吗，如果是正常的张量，应该是不需要L2 norm吧
